cms/auth/wagtail_hooks.py

```diff
@@ -0,0 +1,30 @@
+from django.conf import settings
+from django.utils.html import format_html
+from django.utils.safestring import SafeString, mark_safe
+from django_jinja.builtins.filters import static
+from wagtail import hooks
+
+from cms.auth.utils import get_auth_config
+
+
+def register_global_admin_auth_js_hook():
+    """Conditionally register the global admin Auth JS hook based on an whether Cognito login is enabled."""
+    if not settings.AWS_COGNITO_LOGIN_ENABLED:
+        return
+
+    @hooks.register("insert_global_admin_js")
+    def global_admin_auth_js() -> SafeString:
+        """Register the auth.js script in the Wagtail admin to handle refresh logic."""
+        return format_html(
+            """
+            <script>
+                window.authConfig = {config};
+            </script>
```

Comment:
Issue: This should use the `json_script` template filter, rather than injecting the config in like this, to avoid injection and escaping issues.

Reply:
Great spot! Done!

Reply:
Closes the XSS hole and passes CSP/ XSS audit.

I added deferred-loading auth.js speeds things up.